lib/generate.js

```diff
@@ -80,6 +81,26 @@ function clearPriorOutput(directory, sourceImageBasename) {
   return del(path.join(directory, `${extlessBasename}*.*`));
 }
 
+/**
+ * Handle optimizing file(s)
+ *
+ * @param {string} options
+ * @param {array} results
+ * @return {array} - Resolves with an array of optimized filenames.
+ */
+function handleOptimize(options, results) {
+  if (options.maxConcurrency) {
+    const limit = pLimit(options.maxConcurrency);
+    const optimizeQueue = [];
+    for (const res in results) {
+      optimizeQueue.push(
+        limit(() => optimize(_.flatten(results[res]), options))
+      );
+    }
+    return Promise.all(optimizeQueue);
+  } else return optimize(_.flatten(results), options);
```

Comment:
Do you need `{}` around the return? I've always done it that way if the `if` includes `{}`, but maybe it's not necessary. Like this?

```suggestion
  } else {
    return optimize(_.flatten(results), options);
  }
```

Reply:
@colleenmcginnis It isn't necessary if the value is all on one line, but I can add the curly brackets to improve legibility!